### PR TITLE
connected to redux and created eslint rules

### PR DESCRIPTION
**Pull request by Ksenaaa**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,38 +12,25 @@
     "@types/node": "^18.11.9",
     "@types/react": "^18.0.25",
     "@types/react-dom": "^18.0.9",
+    "@types/react-redux": "^7.1.24",
     "classnames": "^2.3.2",
     "node-sass": "^8.0.0",
```
Comment (valentinPortside) [inline package.json]: shouldn't it be in devDependencies

File: src/components/Loading/Loading.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+
+import styles from './Loading.module.scss'
+
+export const Loading = () => {
```
Comment (valentinPortside) [inline src/components/Loading/Loading.tsx]: rename pls to 'Loader'

File: src/components/Slider/Slider.tsx [hunk 3]
```diff
@@ -1,63 +1,64 @@
...
+    }, [dispatch, sliderPictures?.length])
 
-    const nextPicture = () => {
-        setIndexPicture(preState => pictures.length-1 === preState ? 0 : preState + 1)
-    }
-
-    const previousPicture = () => {
-        setIndexPicture(preState => preState === 0 ? pictures.length-1 : preState - 1)
-    }
+    const onClickPrevious = useCallback(() => {
```
Comment (valentinPortside) [inline src/components/Slider/Slider.tsx]: const onClickPrevious = useCallback(() => dispatch(previousPicture(sliderPictures?.length)), [dispatch, sliderPictures?.length])

File: src/store/counterSliderSlice.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import { createSlice } from '@reduxjs/toolkit'
```
Comment (valentinPortside) [inline src/store/counterSliderSlice.ts]: move this logic to inside Slider component

**Review by valentinPortside — APPROVED** [review]

File: src/store/services/slider.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { IImgSlider } from 'interface/IImgSlider'
+
+import { api } from './api'
+
+export const sliderApi = api.injectEndpoints({
+    endpoints: (builder) => ({
+        getItems: builder.query<IImgSlider[], void>({
```
Comment (valentinPortside) [inline src/store/services/slider.ts]: getSliderImgs

File: src/store/services/slider.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+            providesTags: (result = []) => [
+                ...result.map(({ id }) => ({ type: 'Slider', id } as const)),
+                { type: 'Slider' as const, id: 'LIST' }
+            ]
+        })
+    })
+})
+
+export const {
+    useGetItemsQuery
```
Comment (valentinPortside) [inline src/store/services/slider.ts]: useSliderImgs

File: package.json (hunk 1 above)
Comment (Ksenaaa) [reply package.json]: fixed

File: src/components/Loading/Loading.tsx (hunk 2 above)
Comment (Ksenaaa) [reply src/components/Loading/Loading.tsx]: fixed

File: src/components/Slider/Slider.tsx (hunk 3 above)
Comment (Ksenaaa) [reply src/components/Slider/Slider.tsx:28]: the type of functions must be the same

Comment (Ksenaaa) [reply src/components/Slider/Slider.tsx]: fixed

File: src/store/counterSliderSlice.ts (hunk 4 above)
Comment (Ksenaaa) [reply src/store/counterSliderSlice.ts]: fixed

File: src/store/services/slider.ts (hunk 5 above)
Comment (Ksenaaa) [reply src/store/services/slider.ts]: fixed

File: src/store/services/slider.ts (hunk 6 above)
Comment (Ksenaaa) [reply src/store/services/slider.ts]: fixed

File: src/components/Slider/Slider.tsx [hunk 7]
```diff
@@ -2,62 +2,62 @@ import React, { useCallback, useEffect, useRef, useState } from 'react'
...
         if (!refContainer?.current) return
         const child = refContainer?.current as HTMLDivElement
+
         child?.children[num]?.scrollIntoView({ behavior: 'smooth', block: 'start' })
     }
 
     useEffect(() => {
-        const timeNext = setTimeout(() => nextPicture(), 3000)
+        const timeNext = setTimeout(() =>
+            sliderPictures?.length && setIndexPicture(preState =>
```
Comment (vell28) [inline src/components/Slider/Slider.tsx]: if (!sliderPictures?.length) return 

above setTimeout

**Comment by vell28** [comment]
Fix name of interface files

File: src/components/Slider/Slider.tsx (hunk 7 above)
Comment (Ksenaaa) [reply src/components/Slider/Slider.tsx]: fixed

**Review by vell28 — APPROVED** [review]

**Pull request merged**